lib/PipeboxCommon/Python/_pipebox.cpp

```diff
@@ -86,7 +86,11 @@ NB_MODULE(_pipebox, m) {
            return Handle.diff(*nanobind::cast<Model *>(Other));
          })
     .def("clone", &Model::clone)
-    .def("serialize", &Model::serialize)
+    .def("serialize",
+         [](Model &Handle) {
+           llvm::SmallVector<char, 0> Buffer = Handle.serialize();
+           return nanobind::bytes(Buffer.data(), Buffer.size());
```

Comment:
Is this copying?

Reply:
Yes, but there's no way around it, if you return `bytes` then it needs to be copied from C

Reply:
I understand, I see in the CPython docs there's no "move contructor".

Add a comment about the fact that we're copying.

> model_yaml = yaml.safe_load(bytes(model.serialize()))

`yaml.safe_load` takes a `stream` as far as I can tell.

```
Help on function safe_load in module yaml:

safe_load(stream)
    Parse the first YAML document in a stream
    and produce the corresponding Python object.

    Resolve only basic YAML tags. This is known
    to be safe for untrusted input.
```

Maybe we can use use `BytesIO` with a `memoryview`?
We don't need to do this now but, if viable, it's worth to leave a comment about this could be done without a copy.

Reply:
Added a `TODO` explaining the situation